### PR TITLE
Get indexInfo and segment size by queryCoord

### PR DESCRIPTION
**Pull request by xige-16**
fix issue: #14225 
/kind enhancement
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/querycoord/util.go [hunk 1]
```diff
@@ -30,15 +31,47 @@ func getCompareMapFromSlice(sliceData []int64) map[int64]struct{} {
...
+
+	// get size of delete data
+	for _, fieldBinlog := range segmentLoadInfo.Deltalogs {
+		segmentSize += GetFieldSizeFromFieldBinlog(fieldBinlog)
+	}
+
+	return segmentSize
+}
+
+func GetFieldSizeFromFieldBinlog(fieldBinlog *datapb.FieldBinlog) int64 {
```
Comment (bigsheeper) [inline internal/querycoord/util.go:67]: maybe shouldn't be exported, `getFieldSizeFromFieldBinlog`

**Comment by jennyli-z** [comment]
manually run passed on qa cluster twice.

File: internal/querycoord/util.go (hunk 1 above)
Comment (xige-16) [reply internal/querycoord/util.go]: fixed

File: internal/querycoord/global_meta_broker.go [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+	if pathResponse.Status.ErrorCode != commonpb.ErrorCode_Success {
+		err = fmt.Errorf("get index info from indexCoord failed, buildID = %d, reason = %s", buildID, pathResponse.Status.Reason)
+		log.Error(err.Error())
+		return nil, err
+	}
+
+	return pathResponse.FilePaths, nil
+}
+
+func (broker *globalMetaBroker) setIndexInfo(ctx context.Context, segmentID UniqueID, indexInfo *querypb.VecFieldIndexInfo) error {
```
Comment (xiaofan-luan) [inline internal/querycoord/global_meta_broker.go:190]: parseIndexInfo should be better?

**Comment by xiaofan-luan** [comment]
overall is good, looks much clear than before

File: internal/querycoord/global_meta_broker.go (hunk 2 above)
Comment (xige-16) [reply internal/querycoord/global_meta_broker.go]: fixed

File: internal/querycoord/global_meta_broker.go [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+			zap.Int64("segmentID", segmentID),
+			zap.Error(err))
+		return false, 0, err
+	}
+
+	if !response.EnableIndex {
+		return false, 0, nil
+	}
+
+	return true, response.BuildID, nil
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querycoord/global_meta_broker.go:182]: fixed

File: internal/querynode/segment.go [hunk 4]
```diff
@@ -30,6 +30,7 @@ import (
 	"encoding/binary"
 	"errors"
 	"fmt"
+	"github.com/milvus-io/milvus/internal/proto/querypb"
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querynode/segment.go]: fixed

File: internal/querycoord/global_meta_broker.go [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+
+	dataKV, err := minioKV.NewMinIOKV(childCtx, option)
+	if err != nil {
+		return nil, err
+	}
+	parser.dataKV = dataKV
+	return parser, nil
+}
+
+func (broker *globalMetaBroker) stopProxySendQueryMsg(ctx context.Context, collectionID UniqueID) error {
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querycoord/global_meta_broker.go]: fixed

File: internal/querycoord/global_meta_broker.go [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+		log.Error("releaseDQLMessageStream occur error", zap.Int64("collectionID", collectionID), zap.Error(err))
+		return err
+	}
+	if res.ErrorCode != commonpb.ErrorCode_Success {
+		err = errors.New(res.Reason)
+		log.Error("releaseDQLMessageStream occur error", zap.Int64("collectionID", collectionID), zap.Error(err))
+		return err
+	}
+
+	return nil
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querycoord/global_meta_broker.go:84]: fixed

File: internal/querycoord/global_meta_broker.go [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+	getRecoveryInfoRequest := &datapb.GetRecoveryInfoRequest{
+		Base: &commonpb.MsgBase{
+			MsgType: commonpb.MsgType_GetRecoveryInfo,
+		},
+		CollectionID: collectionID,
+		PartitionID:  partitionID,
+	}
+	recoveryInfo, err := broker.dataCoord.GetRecoveryInfo(ctx2, getRecoveryInfoRequest)
+	if err != nil {
+		return nil, nil, err
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querycoord/global_meta_broker.go:125]: fixed

File: internal/querycoord/global_meta_broker.go [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+	}
+
+	if recoveryInfo.Status.ErrorCode != commonpb.ErrorCode_Success {
+		err = errors.New(recoveryInfo.Status.Reason)
+		return nil, nil, err
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querycoord/global_meta_broker.go:131]: fixed

File: internal/querynode/load_index_info.go [hunk 9]
```diff
@@ -25,6 +25,8 @@ package querynode
 */
 import "C"
 import (
+	"github.com/milvus-io/milvus/internal/proto/querypb"
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querynode/load_index_info.go]: fixed

**Pull request merged**